modyn/trainer_server/internal/grpc/trainer_server_grpc_servicer.py

```diff
@@ -355,24 +362,30 @@ def store_final_model(
                 trigger_id=trigger_id,
                 hostname=self._config["trainer_server"]["hostname"],
                 port=int(self._config["trainer_server"]["ftp_port"]),
-                model_path=prefix_path,
+                model_path=str(prefix_model_path),
+                checksum=calculate_checksum(final_checkpoint_path),
             )
 
             register_response: RegisterModelResponse = self.model_storage_stub.RegisterModel(register_request)
 
             if not register_response.success:
-                logger.error(f"Could not store final model from training id {training_id}.")
+                logger.error(f"Could not store final model from training id {training_id} at model storage.")
                 return StoreFinalModelResponse(valid_state=False)
-
-            os.remove(final_model_path)
-
-            logger.info(f"Deleted final model at {final_model_path}")
+            os.remove(final_checkpoint_path)
+            logger.info(f"Deleted final model on path {final_checkpoint_path}")
 
             return StoreFinalModelResponse(valid_state=True, model_id=register_response.model_id)
 
         logger.error(f"Could not find final checkpoint of training with ID {training_id}.")
         return StoreFinalModelResponse(valid_state=False)
 
+    def _prepare_final_model(self, training_id: int) -> Optional[pathlib.Path]:
```

Comment:
i don't get that function name yet. what is being prepared? this is more like a path "calculation"?

Reply:
You are right, this is misleading. I changed the name.